packages/fast-components-react-msft/src/button/button.tsx

```diff
@@ -56,26 +56,14 @@ class Button extends Foundation<ButtonHandledProps, ButtonUnhandledProps, {}> {
      * Generates class names
      */
     protected generateClassNames(): string {
-        switch (this.props.appearance) {
-            case ButtonAppearance.primary:
-                return super.generateClassNames(
-                    get(this.props, "managedClasses.button__primary")
-                );
-            case ButtonAppearance.outline:
-                return super.generateClassNames(
-                    get(this.props, "managedClasses.button__outline")
-                );
-            case ButtonAppearance.lightweight:
-                return super.generateClassNames(
-                    get(this.props, "managedClasses.button__lightweight")
-                );
-            case ButtonAppearance.justified:
-                return super.generateClassNames(
-                    get(this.props, "managedClasses.button__justified")
-                );
-            default:
-                return super.generateClassNames();
-        }
+        const className: string = this.props.appearance
+            ? get(
+                  this.props,
+                  `managedClasses.button__${ButtonAppearance[this.props.appearance]}`
+              )
+            : "";
```

Comment:
nit: It would be more accurate to use set this empty string as `undefined` and the typing can be `string | void`.